internal/reconcilers/run_profile.go

```diff
@@ -78,14 +77,8 @@ func (r *Reconciler) handleProfileInitEvent(msg *message.Message) error {
 		return nil
 	}
 
-	ectx := &engine.EntityContext{
```

Comment:
The use of this type was leading to a circular dependency, and after reviewing the code - it appears it is only used to pass around a single value. Instead of fixing the circular dependency, I just removed the use of this type here.